### ISSUE TITLE
[Expo Go] [Android Only] React native build error: Attempt to invoke virtual method'boolean com.facebook.react.uimanager.FabricViewStateManager.hasStateWrappper()

### ISSUE THREAD

**Issue by BrodaNoel** [expo/expo#17372]
## 💡 Note from Expo team:

Please refer to https://github.com/expo/expo/issues/17372#issuecomment-1119866941 for a workaround (this explains how to install an older version of Expo Go if you are still using SDK 44). Another option is to [upgrade to SDK 45](https://blog.expo.dev/expo-sdk-45-f4e332954a68). We are working towards a resolution for the latest version.

<hr />

### Summary

I just updated Expo Go to the last version (getting ready for SDK 45).
iOS working well, but in Android I have exactly this issue: https://stackoverflow.com/questions/67683149/react-native-build-error-attempt-to-invoke-virtual-methodboolean-com-facebook

I don't have any error log in the console to share here. Just same screenshot as that SO question.
It happens right exactly when the app finish loading.

I can see in the logs that it says something about "emoji". The only thing I can say is that my app DOES HAVE emojis.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

SDK44

### Environment

  expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 11.6.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.18.2 - ~/.nvm/versions/node/v14.18.2/bin/node
      Yarn: 1.22.18 - ~/.yarn/bin/yarn
      npm: 8.3.2 - ~/.nvm/versions/node/v14.18.2/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 27, 31
        Build Tools: 23.0.1, 23.0.3, 25.0.3, 26.0.1, 27.0.3, 28.0.3, 29.0.2
        System Images: android-31 | Google Play Intel x86 Atom_64
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8309675
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.6
      react: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
    npmGlobalPackages:
      eas-cli: 0.52.0
      expo-cli: 5.4.3
    Expo Workflow: managed

### Reproducible demo

1:50AM here. Kind of hard to get a reproducible demo now.

**Comment by BrodaNoel** [comment]
Probably duplicated from https://github.com/expo/expo/issues/17335 and https://github.com/facebook/react-native/issues/31572#issuecomment-912017067

The weird thing is that this issue was not present there in the previous version of Expo Go.

**Comment by tingxiao69** [comment]
Same issue here, don't know how to specify the 
```
implementation ("androidx.appcompat:appcompat:1.3.1") {
        version {
            strictly '1.3.1'
        }
    }
```
in expo. If there is a way, then this issue will be solved for me

**Comment by MarcosKerchner** [comment]
I'm having the same problem with the <TextInput>

**Comment by MPBogdan** [comment]
Same issue here. For Android I just downloaded the previous version of apk (2.23.2) from apkpure. I have this error only with the latest version 2.24.1.

**Comment by leonardoartuso88** [comment]
Também tive esse problema ao atualizar o expo, irei baixar a versão 2.23.2 conforme @MPBogdan e ver se resolve.

**Comment by adam-adair** [comment]
Same issue with 2.24.1

**Comment by Chagall** [comment]
> Same issue here. For Android I just downloaded the previous version of apk (2.23.2) from apkpure. I have this error only with the latest version 2.24.1.

I did the same thing and can confirm that my app is working again on Expo 2.32.2. It must be a bug on the version 2.24.1 of expo go.

**Comment by samuthekid** [comment]
Same here

**Comment by Sohail-panchoo** [comment]
> Same issue here. For Android I just downloaded the previous version of apk (2.23.2) from apkpure. I have this error only with the latest version 2.24.1.

ive used previous version of expo go
version 2.23.2 you can download from apkpure this solved the problem for me

**Comment by Ghassanee** [comment]
Happened to me when I tired to work with TextInput 
weird bug 
anyway I downgraded the expo go version to v2.23.2

**Comment by vitorhonna** [comment]
Same issue, solved downgrading the expo go version to v2.23.2

**Comment by johntravolta1** [comment]
Same. Solved with version 2.23.2

**Comment by babacarMbengue12** [comment]
Same issue on android after upgrading Expo Go

**Comment by alexrsaff** [comment]
Same issue here. Expo Go updated automatically when I stepped away for a sec since the phone was plugged in and charging from my laptop.

**Comment by tsapeta** [comment]
Hi guys, thanks for reporting this issue! 🙏
We're investigating the problem and will let you know about any updates.

**Comment by johnphealipto** [comment]
Same here, 
Just updated Expo Go and all screens with the TextInput component kept crashing...
Probably have to downgrade Expo Go to v2.23.2 till the issue is fixed

**Comment by byCedric** [comment]
While we investigate this issue, I want to mention some easy ways to get back to the previous Expo Go version. Expo CLI has a command to install any specific version of Expo Go on your local device or simulator.

- `$ expo client:install:android --device`
- `$ expo client:install:ios --device`

If you run this inside a project, it will list the available Expo Go versions for that project. If you run this outside your project, you can pick any Expo Go version you want.

![image](https://user-images.githubusercontent.com/1203991/167194396-12a89a52-0d68-463b-a587-6353a638dd9f.png)

Sorry for the inconvenience!

**Comment by byCedric** [comment]
To anyone running into this issue, can someone provide a repro example? It looks like this might be coming from one or more 3rd party modules. [According to this comment](https://github.com/microsoft/react-native-test-app/pull/643/files#diff-4b292ae09e1b1f683e8ef999c99a99fbc34cccf51d7738528fb88ae25ea0d04aR24-R27), it's likely caused by an issue in React Native and `androidx.appcompat:appcompat:1.4.0+`.

[In our modules, we only use `1.2.0`, `1.1.0`, or `1.0.2`](https://github.com/expo/expo/search?p=1&q=androidx.appcompat%3Aappcompat).

**Comment by borasumer** [comment]
> To anyone running into this issue, can someone provide a repro example? It looks like this might be coming from one or more 3rd party modules. [According to this comment](https://github.com/microsoft/react-native-test-app/pull/643/files#diff-4b292ae09e1b1f683e8ef999c99a99fbc34cccf51d7738528fb88ae25ea0d04aR24-R27), it's likely caused by an issue in React Native 0.68.x and `androidx.appcompat:appcompat:1.4.0+`.
> 
> [In our modules, we only use `1.2.0`, `1.1.0`, or `1.0.2`](https://github.com/expo/expo/search?q=androidx.appcompat%3Aappcompat%3A).

This seems like a TextInput issue for me on Android, that is imported from 'react-native'. App works if I comment the TextInput out, on any screen.

**Comment by brentvatne** [comment]
@borasumer - can you prepare a minimal reproducible example for us? https://stackoverflow.com/help/minimal-reproducible-example

**Comment by sankparatkar** [comment]
@byCedric  how can I connect my device before running the commands you shared? I'm not able to understand how to

**Comment by byCedric** [comment]
@sankparatkar You have to connect your device to ADB in this case, so USB connected and authorized to develop.
- [Setup USB debugging on your device](https://developer.android.com/studio/debug/dev-options#enable)
- Connect your device to your computer with a cable
- Start ADB (or `expo start --android`)
- Run the `expo client:install:android --device` command in a separate terminal

**Comment by TheBorgess** [comment]
I just reinstalled the old version 2.23.2, and everything went ok.

**Comment by utsavbhattarai007** [comment]
same issue.
install older version to solve it

**Comment by mikemasam** [comment]
This issue has been resolved on react native https://github.com/facebook/react-native/commit/e21f8ec34984551f87a306672160cc88e67e4793
 
Upgrade to Expo 45

**Comment by philemonpeter02** [comment]
Facing the same issue on every screen that has a textInput im using TextInput and TextArea from native base

**Comment by azarsammie** [comment]
I am also facing the same Issue

**Comment by TalalNuman** [comment]
Getting the same issue, after using react native's text input.

**Comment by tvbegovic** [comment]
Tried to upgrade to Expo SDK 45 with expo upgrade but I'm getting this error during upgrade process:
Failed to install expo package with error: yarnpkg exited with non-zero code: 134
![image](https://user-images.githubusercontent.com/8981787/167246826-0291ab83-7cf8-42c2-ba04-8162e17049f1.png)

**Comment by turtlefung** [comment]
same issue, seems like the textinput imported from "react-native" is the cause of the problem

**Comment by VlasisMentz** [comment]
Same problem here. I can't use TextInput from 'react-native' and i already updated Expo Go.

**Comment by jroux-git** [comment]
I can confirm that this solution by byCedric is working [solution](https://github.com/expo/expo/issues/17372#issuecomment-1119866941)

I use Expo44 and downgraded to 2.32.2 of Expo Go and all is good again.

**Comment by bdhcode** [comment]
Me too everything works fine yesterday but today the app crashs on screen having textinput

**Comment by bdhcode** [comment]
@byCedric thank you for the temporary solution. Guys use this solution until the fix is released

**Comment by kris1803** [comment]
Initialising expo init with all updated, with android and win 10, having issues when bundling for android device with expo go. SDK 45. It happens when I add router navigation, precisely router tab navigator. SDK 45 (anything was working 1 week ago).

**Comment by zehratok** [comment]
Same. Solved with version 2.23.2

**Comment by zehratok** [comment]
solved with version 2.23.2

**Comment by aalzubidy** [comment]
Having same issue on Android with expo go 2.24.2

**Comment by brentvatne** [comment]
#17392 fixes this, we're working on rolling this out now

**Comment by brentvatne** [comment]
the fix is rolled out to 2.24.3, soon to be available on the google play store and currently the default version of expo go installed through expo-cli. you can also download the apk directly here: https://d1ahtucjixef4r.cloudfront.net/Exponent-2.24.3.apk
